### PR TITLE
Prepare for export so that other libraries can extend

### PR DESCRIPTION
**Pull request by SparshithNR**
Update the export so that this library can run as node module

Update the README with usage and contribution
README Rendered: https://github.com/stefanpenner/supported/blob/6cfdcf420c362227fb29e1343c27b53a80311aa5/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,115 @@
...
+			url: 'http://custom-npm-registry.com',
+			type:  'dependency',
+		});
+	};
+	return {
+		dependenciesToCheck,
+		pkg: packageJson,
+	}
+}
+const  jsonResult = await isAllInSupportWindow(projectPaths, setupProject);
```
Comment (stefanpenner) [inline README.md:73]: nit: `. ` extra whitespace

File: README.md [hunk 2]
```diff
@@ -1,8 +1,115 @@
...
+You can extend the CLI and keep the same option, want to change the name of the tool then do the following,
+```js
+// bin/custom-support-cli
+#!/usr/bin/env node
+const { run } = require('supported');
+const  helpFn = require('supported/lib/help');
+function setupProjectCustom(projectPath) {
+	let packageJson = JSON.parse(fs.readFileSync(`${projectPath}/package.json`, 'utf-8'));
+	let dependenciesToCheck = [];
+	for (const [name, version] of  Object.entries(pkg.dependencies) {
```
Comment (stefanpenner) [inline README.md:84]: nit: `. ` extra whitespace

File: README.md [hunk 3]
```diff
@@ -1,8 +1,115 @@
...
 ```sh
 npx supported <path/to/node_module>
+npx supported <[array/of/node_modules]>
+```
+
+### As a node module
+
+
+```js
+const { isAllInSupportWindow } = require('supported');
```
Comment (stefanpenner) [inline README.md:19]: can we nit-pick this API name?

* `is` to be implies, true/false return value, but we get a complex structure back
* `all` all of what?

The way I think about a function name:
A function name is the relationship between the functions inputs, and the functions (outputs, or side-effects).

Given:

```js
let b = isAllInSupportWindow(a);
```

I read this as  `B` will represent true or false based on if all of `A` is in the support window. Rather, it appears that the relationship between `A` and `B` is more `B` is the result of running support policies across `A`. And then B can be inspected for the result of the policies.

Given this, any ideas on an improved function name?

File: README.md [hunk 4]
```diff
@@ -1,8 +1,115 @@
...
+const projectPaths = ['test/fixtures/unsupported-project', 'test/fixtures/supported-project' ];
+const jsonResult = await isAllInSupportWindow(projectPaths);
+const projectPaths_2 = 'test/fixtures/unsupported-project';
+const jsonResult_2 = await isAllInSupportWindow(projectPaths_2);
+```
+
+
+
+Developers can pass custom setupProjectFn which will return a object as below
```
Comment (stefanpenner) [inline README.md:28]: When would custom developers need to/want to use that?

File: README.md [hunk 5]
```diff
@@ -1,8 +1,115 @@
...
+
+```js
```
Comment (stefanpenner) [inline README.md:30]: This example likely needs more polish as it is hard to follow.
If you are in a rush, maybe we remove it for now and add this back at a later time once there is more time to iterate on it.

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (SparshithNR) [reply README.md:28]: This custom stuff will be when `npm registry` is not the right place to get the artifactory information for a given project. And avoid all the setupProject steps we provide by default.

File: README.md (hunk 3 above)
Comment (SparshithNR) [reply README.md:19]: Can we rename it to `getSupportPolicyResult`?

File: README.md (hunk 5 above)
Comment (SparshithNR) [reply README.md]: I updated the readme with minimum items. We will update the polished readme later.

**Review by stefanpenner — APPROVED** [review]
